include/bactria/Event.hpp

```diff
@@ -13,62 +13,180 @@
+             * the generic Category.
+             */
             Event() : Marker("BACTRIA_GENERIC_EVENT", bactria::color::orange, Category{})
             {}
 
+            /**
+             * \brief The constructor.
+             *
+             * The constructor will generate an event with the name \a name, the color \a color and the Category \a
+             * category.
+             *
+             * \param name The name as it should appear on the visualizer.
+             * \param color The color as it should appear on the visualizer. Needs to be supplied in ARGB format.
+             * \param category The category this event should be assigned to. This allows for later filtering.
+             */
             Event(std::string name, std::uint32_t color = bactria::color::orange, Category category = Category{})
             : Marker(std::move(name), color, std::move(category))
             {}
 
+            /**
+             * \brief The copy constructor.
+             *
+             * The copy constructor will generate a new event with the same properties as the original.
+             *
+             * \param other The Event to copy the properties from.
+             */
             Event(const Event& other)
             : Marker(other)
-            , m_handle{plugin::create_event(m_name.c_str(), m_color, m_category.id)}
+            , m_handle{detail::is_activated() ? plugin::create_event(m_color, m_category.get_c_name(), m_category.get_id()) : nullptr}
+            , m_action{other.m_action}
             {}
 
+            /**
+             * \brief The copy assignment operator.
+             *
+             * The copy assignment operator will generate a new event with the same properties as the original.
+             *
+             * \param rhs The Event to copy the properties from.
+             */
             auto operator=(const Event& rhs) -> Event&
             {
                 Marker::operator=(rhs);
-                m_handle = plugin::create_event(m_name.c_str(), m_color, m_category.id);
+
+                if(detail::is_activated())
+                    m_handle = plugin::create_event(m_color, m_category.get_c_name(), m_category.get_id());
+                else
+                    m_handle = nullptr;
+
+                m_action = rhs.m_action;
 
                 return *this;
             }
             
-            Event(Event&& other) noexcept = default;
-            auto operator=(Event&& rhs) noexcept -> Event& = default;
+            /**
+             * \brief The move constructor.
+             *
+             * The move constructor will move the contents of the original Event into this Event. The original
+             * Event will be in an invalid state; using it is undefined behaviour.
+             *
+             * \param other The Event to be moved.
+             */
+            Event(Event&& other) noexcept
+            : Marker(std::forward<Event>(other)), m_action{std::move(other.m_action)}
```

Comment:
`Event&&` is not a forwarding reference here, because `Event` is not a deduced template parameter of the function. Therefore you do not need `std::forward`, which is equal to a `std::move` here anyway, but just a `std::move`. It feels weird though to move `other` and then also move the action of `other` afterwards. I kind of violates the don't touch objects after move rule. But it might actually be correct. I honestly don't know :/

Reply:
"The internet" recommends `std::move`. I will change this.